src/NHibernate/Transaction/AdoNetWithDistributedTransactionFactory.cs

```diff
@@ -69,6 +69,9 @@ public void EnlistInDistributedTransactionIfNeeded(ISessionImplementor session)
 						}
 						session.ConnectionManager.AfterTransaction();
 						session.AfterTransactionCompletion(wasSuccessful, null);
+						foreach (var dependentSession in session.ConnectionManager.DependentSessions)
+							dependentSession.AfterTransactionCompletion(wasSuccessful, null);
+
```

Comment:
That was the path I had taken before pushing my PR, but duplicating that logic in transaction factory and ado transaction was not looking good to me.